### PR TITLE
VEX support

### PR DESCRIPTION
**Pull request by slashben**
## PR Type:
Enhancement, Other

___
## PR Description:
This PR introduces the following changes:
- Adds support for VEX (Vulnerability Exchange) by implementing cleanup of VEX documents in storage.
- Updates the 'go-cmp' dependency from v0.5.9 to v0.6.0.
- Replaces the 'kubescape/storage' dependency with a specific version.

___
## PR Main Files Walkthrough:
<details> <summary>files:</summary>

`watcher/watcher.go`: Added a check to delete OpenVulnerabilityExchangeContainers if the instance ID is not recognized. This is part of the VEX support implementation.
`go.mod`: Updated the 'go-cmp' dependency version and replaced the 'kubescape/storage' dependency with a specific version.
`go.sum`: Updated the checksums for the updated and replaced dependencies.
</details>

___
## Overview

Adding support for VEX: cleanup of VEX document in storage. Part of this feature https://github.com/kubescape/kubevuln/issues/155

**[Signed Commits](../CONTRIBUTING.md#sign-off-per-commit)**
- [X] Yes, I signed my commits.

### CODE REVIEW

**Comment by slashben** [comment]
> @slashben is this ready?

Still waiting for @vladklokun to review

File: watcher/watcher.go [hunk 1]
```diff
@@ -282,6 +282,18 @@ func (wh *WatchHandler) HandleSBOMFilteredEvents(sfEvents <-chan watch.Event, pr
 			continue
 		}
 
+		if !slices.Contains(wh.managedInstanceIDSlugs, hashedInstanceID) {
+			wh.storageClient.SpdxV1beta1().OpenVulnerabilityExchangeContainers(obj.ObjectMeta.Namespace).Delete(context.TODO(), obj.ObjectMeta.Name, v1.DeleteOptions{})
```
Comment (dwertent) [inline watcher/watcher.go:286]: Run this flow only when the VEX capability is enabled

**Review by dwertent — CHANGES_REQUESTED** [review]

File: watcher/watcher.go (hunk 1 above)
Comment (slashben) [reply watcher/watcher.go:286]: If VEX capability is not enabled then there are no VEX objects to delete. I don't want to pollute the code by adding `if`-s all around these components. The nice thing that the whole system is object driven, if the object not there nothing happens

**Comment by dwertent** [comment]
We manage the delete in the storage

**Pull request closed without merging**